alg/sha256.c

```diff
@@ -17,13 +17,13 @@
     defined(CPUSUPPORT_ARM_SHA256)
 #define HWACCEL
 
-enum {
+static enum {
```

Comment:
If you add appropriate `#ifdef`s here, do you remove the need for `-Wno-switch-enum`?

Reply:
Technically yes we would avoid that need, but then we'd get warnings about unused macros (due to other platforms).  Hence 37237daa3cf7deaead9d1746b85397e5c7e049d6 a month ago.

What we _could_ do is:
```
#define HW_UNSET 0
#define HW_SOFTWARE 1
#if defined(CPUSUPPORT_X86_SHANI) && defined(CPUSUPPORT_X86_SSSE3)
#define HW_X86_SHANI 2
#endif
#if defined(CPUSUPPORT_X86_SSE2)
#define HW_X86_SSE2 3
#endif
#if defined(CPUSUPPORT_ARM_SHA256)
#define HW_ARM_SHA256 4
#endif
```

... but that looks so ugly that I didn't bother proposing it.

On the other hand, sha256 is the worst case (in terms of number of platforms), and this code only needs to be written once.  So I'm happy to rewrite the PR to use this, if you want.

Reply:
Did you mean 2d664b8b?

Reply:
I was thinking we could have

```
static enum {
    HW_SOFTWARE,
#if defined(CPUSUPPORT_X86_SHANI) && defined(CPUSUPPORT_X86_SSSE3)
    HW_X86_SHANI,
#endif
#if defined(CPUSUPPORT_X86_SSE2)
    HW_X86_SSE2,
#endif
#if defined(CPUSUPPORT_ARM_SHA256)
    HW_ARM_SHA256,
#endif
    HW_UNSET
} hwgood;
```

Reply:
Whoops, yeah, that's the commit I meant.

I could have sworn that it didn't work, but as long as we don't have a `default` inside the switch, it looks ok.